text/0096-metric-instrument-terminology.md

```diff
@@ -0,0 +1,153 @@
+"Measure":
+
+1. OTEP 88 stipulates that the terms currently in use to name
+synchronous and asynchronous instruments become abstract, but also
+using "Measure-like" and "Observer-like" to discuss instruments with
+refinements.  This proposal states that we shall prefer the
+adjectives, commonly abbreviated "Sync" and "Async", when describing
+instruments.
+2. Prior to OTEP 88, but even with OTEPs 88 and 93 included, there is
+inconsistency in the naming of instruments.  Note that "Counter" and
+"Observer" end in "-er", a noun suffix used in the sense of "[person
+occupationally connected
+with](https://www.merriam-webster.com/dictionary/-er)", while the term
+"Measure" does not fit this pattern.  This proposal proposes to
+replace the abstract term "Measure" by "Recorder", since the
+associated method name (verb) is specified as `Record()`.
+3. The OTEP 93 asynchronous instruments ("LastValueObserver",
+"DeltaObserver", and "CumulativeObserver") have the pattern
+"-Observer", while the OTEP 93 synchronous instruments
+("Counter", "UpDownCounter", "Distribution", "Timing") do not.  This
+proposal keeps "Counter" and "UpDownCounter" for Sum-only synchronous instruments, and does the same
+with "Recorder", yielding "Recorder" and "TimingRecorder".
+4. Confusion over the loss of "Gauge" is addressed by replacing
+"LastValueObserver" with "GaugeObserver".
+
+This proposal also repeats the current specification of the default
+Aggregator for each kind of instrument.
+
+## Explanation
+
+The following table summarizes the four synchronous instruments and
+three asynchronous instruments that will be standardized as a result
+of this set of proposals.
+
+| Existing name | OTEP 93 name       | **Final name**         | Sync or Async | Function | Default aggregation | Rate support |
+| ------------- | ------------------ | ---------------------- | ----------- | ------------- | ---------- | ---- |
+| Counter       | Counter            | **Counter**            | Sync  | Add() | Sum | Yes | 
+|               | UpDownCounter      | **UpDownCounter**      | Sync  | Add() | Sum | Yes |
+| Measure       | Distribution       | **Recorder**           | Sync  | Record() | MinMaxSumCount | No |
+|               | Timing             | **TimingRecorder**     | Sync  | Record() | MinMaxSumCount  | No |
+| Observer      | LastValueObserver  | **GaugeObserver**      | Async | Observe() | MinMaxSumCount | No |
+|               | DeltaObserver      | **DeltaObserver**      | Async | Observe() | Sum | Yes |
+|               | CumulativeObserver | **CumulativeObserver** | Async | Observe() | Sum | Yes |
+
+The argument for "Recorder" instead of "Distribution" is that we
+should prefer instrument descriptives associated with the action being
+performed ("occupationally connected"), not the value being computed,
+as the latter is dependent on SDK configuration.  A "Recorder" records
+a value that is part of a distribution.  A "Counter" counts a value
```

Comment:
Have we considered other names for the "Recorder", I do understand the argument against "Distribution" but I do not really like "Recorder". "Recorder" is a very generic name and even in this sentence you had to refer to what types of values you record "A "Recorder" records a value that is part of a distribution.". I am worried that "Recorder" will supersede all the others in terms of usage. I would like to hear more opinions/suggestions.

Reply:
Any time there is a set of values, you have a distribution, so I considered "that is part of a distribution" to be almost unnecessary, except there is emphasis being placed on it producing a distribution.  I would expect Counter is the most common instrument, after that I would expect Recorder to be second-most-common.

Reply:
Given the `Recorder` is a foundational instrument I think this generic nature is ideal. It is the base from which refined instruments are created. I think defaulting to a `Recorder` is what we want to happen if you are not sure a refinement matches your intended need.